connect-widget/app/src/pages/connect/components/connect-providers.tsx

```diff
@@ -34,15 +25,18 @@ const ConnectProviders = () => {
     }
 
     getConnectedProviders();
-  }, [searchParams]);
+  }, []);
+  const searchProviders = searchParams.get(Constants.PROVIDERS_PARAM);
 
   return (
     <>
-      <Flex p={4} m={4} justify={"center"}>
-        <Heading textAlign={"center"}>Connect your sources.</Heading>
+      <Flex justify={"center"}>
+        <Heading mb={2} textAlign={"center"}>
+          Connect your sources.
+        </Heading>
       </Flex>
       <Providers
-        providers={providers}
+        providers={getProviders(searchProviders)}
```

Comment:
If I'm not wrong, this will execute `getProviders()` on each render, no? Even though is just using local storage (and not a request over the wire), it still generates processing that could be saved with a memoized version of the providers?